comms/dht/src/crypt.rs

```diff
@@ -66,6 +69,49 @@ pub fn generate_ecdh_secret(secret_key: &CommsSecretKey, public_key: &CommsPubli
     output
 }
 
+fn pad_message_to_base_length_multiple(message: &[u8]) -> Vec<u8> {
+    let n = message.len();
+    // little endian representation of message length, to be appended to padded message,
+    // assuming our code runs on 64-bits system
+    let prepend_to_message = (n as u64).to_le_bytes();
+
+    let k = prepend_to_message.len();
+
+    let div_n_base_len = (n + k) / MESSAGE_BASE_LENGTH;
+    let output_size = (div_n_base_len + 1) * MESSAGE_BASE_LENGTH;
+
+    // join prepend_message_len | message | zero_padding
+    let mut output = prepend_to_message.to_vec();
+    output.append(&mut message.to_vec());
+    output.append(&mut [0u8; 128].to_vec());
+
+    // we have to truncate the vector to output_size
+    output.truncate(output_size);
```

Comment:
This code has 6 heap allocations, try this instead (1 heap allocation):
```suggestion
    let mut output = Vec::with_capacity(output_size);
    output.extend_from_slice(&prepend_to_message);
    output.extend_from_slice(&message);
    output.extend(iter::repeat(0u8).take(output_size - n - k));
```

Reply:
Amazing ! Thank you